WSL/disk-space.md

```diff
@@ -0,0 +1,217 @@
+- **SIZE**: lets us know how much space the device is using.
+- **RO**: Let's us know if the device is read-only (1) or not (0).
+- **TYPE**: Refers to the device type (disk in this case, could also be something like lvm for a logical volume device).
+- **MOUNTPOINTS**: Refers to the current directory on the files system where the block device is located (SWAP is for preconfigured inactive memory so no mountpoint).
+
+### Read-only fallback error
+
+If WSL encounters an error while installing (mounting) a Linux distribution, it may mount the distribution as read-only (RO = 1) as a fallback. If that happens, the distribution may display the following error during startup:
+
+```powershell
+An error occurred mounting the distribution disk, it was mounted read-only as a fallback.
+```
+
+When a distribution is started as read-only, any attempts to write to the filesystem will fail with an error like this:
+
+```bash
+$ touch file
+touch: cannot touch 'file': Read-only file system
+```
+
+To repair a disk mount error in WSL, and restore it back to a usable / writeable state again, you can use the `wsl --mount` command to re-mount the disk with the following steps:
+
+1. Shutdown all WSL distributions by opening PowerShell and entering the command:
+
+    ```powershell
+    wsl.exe --shutdown
+    ```
+
+2. Open PowerShell as administrator (in an elevated command prompt) and enter the mount command, replacing `<path-to-ext4.vhdx>` with the path to the distribution's .vhdx file. For help locating this file, see [How to locate the VHD file and disk path for your Linux distribution](#how-to-locate-the-vhdx-file-and-disk-path-for-your-linux-distribution).
+
+    ```powershell
+    wsl.exe --mount <path-to-ext4.vhdx> --vhd --bare
+    ```
+
+3. Use the `wsl.exe lsblk` command from PowerShell to identify the block device name for the distribution (sd[a-z]) and then enter the following command to repair the disk (replacing `<device>` with the correct block device name, like "sdc"). The `e2fsck` command checks ext4 file systems (the type used by distributions installed with WSL) for errors and repairs them accordingly.
+
+    ```powershell
+    wsl.exe sudo e2fsck -f /dev/<device>
+    ```
+
+4. Once the repair is complete, unmount the disk in PowerShell by entering:
+
+    ```powershell
+    wsl.exe --unmount
+    ```
+
+> [!WARNING]
+> You can use the command: `sudo mount -o remount,rw /` to return a read-only distribution to a usable/writable state, but all changes will be in-memory and so will be lost when the distribution is restarted. We recommend using the steps listed above to mount and repair the disk instead.
+
+## How to locate the .vhdx file and disk path for your Linux distribution
+
+If you are having trouble locating the *.vhdx* file for a Linux distribution, go to the directory path `%LocalAppData%\Packages\`. Each Linux distribution that you have installed includes a subfolder specific to that distribution.
```

Comment:
That's not true since this path can be changed

Reply:
Let's remove that part to avoid confusion

Reply:
Updated